### PR TITLE
Add tests for trailing space in location in delta lake

### PR DESCRIPTION
**Pull request by pajaks**
## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

**Comment by pajaks** [comment]
@ebyhr Could you run with secrets?

**Comment by findepi** [comment]
/test-with-secrets sha=4dc58914b7373e5a3d66f8691baade031c42a9f7

**Comment by findinpath** [comment]
CI hit https://github.com/trinodb/trino/issues/13199

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java [hunk 1]
```diff
@@ -565,6 +565,18 @@ public void testPathColumn()
         }
     }
 
+    @Test
+    public void testTableLocationTrailingSpace()
+    {
+        String tableName = "table_with_space ";
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java:571]: I would recommend appending a space in location, not table name. 
```suggestion
        String tableName = "table_with_space";
```

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java [hunk 2]
```diff
@@ -565,6 +565,18 @@ public void testPathColumn()
...
+
+        assertUpdate(format("CREATE TABLE %s (customer VARCHAR) WITH (location = 's3://%s/%s')", tableName, bucketName, tableName));
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java:573]:
```suggestion
        assertUpdate(format("CREATE TABLE %s (customer VARCHAR) WITH (location = 's3://%s/%s ')", tableName, bucketName, tableName));
```

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeRegisterTableProcedureTest.java [hunk 3]
```diff
@@ -177,6 +177,29 @@ public void testRegisterTableWithDifferentTableName()
...
+    public void testRegisterTableWithTrailingSpaceInLocation()
+    {
+        String tableName = "test_create_table_with_trailing_space_" + randomNameSuffix();
+        String tableLocationWithSpace = tableName + " ";
+
+        assertQuerySucceeds(format("CREATE TABLE %s WITH (location = '%s') AS SELECT 1 as a, 'INDIA' as b, true as c", tableName, tableLocationWithSpace));
+        assertQuery("SELECT * FROM " + tableName, "VALUES (1, 'INDIA', true)");
+
+        String tableLocation = getTableLocation(tableName);
+        assertThat(tableLocation).isEqualTo(tableLocationWithSpace);
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeRegisterTableProcedureTest.java:190]: I would inline `tableLocation` variable.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeRegisterTableProcedureTest.java [hunk 4]
```diff
@@ -177,6 +177,29 @@ public void testRegisterTableWithDifferentTableName()
...
+
+        String registeredTableName = "test_register_table_with_trailing_space_" + randomNameSuffix();
+        assertQuerySucceeds(format("CALL system.register_table('%s', '%s', '%s')", SCHEMA, registeredTableName, tableLocation));
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeRegisterTableProcedureTest.java:193]: I would replace `tableLocation` with `tableLocationWithSpace` to clarify the test purpose.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeRegisterTableProcedureTest.java [hunk 5]
```diff
@@ -177,6 +177,29 @@ public void testRegisterTableWithDifferentTableName()
...
+        assertQuery("SELECT * FROM " + registeredTableName, "VALUES (1, 'INDIA', true)");
+
+        String registeredTableLocation = getTableLocation(tableName);
+        assertThat(registeredTableLocation).isEqualTo(tableLocationWithSpace);
+
+        assertUpdate(format("DROP TABLE %s", registeredTableName));
+        assertUpdate(format("DROP TABLE %s", tableName));
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeRegisterTableProcedureTest.java:200]: nit: No need to `format` method in this case. https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#prefer-string-formatting

Same for `BaseDeltaLakeConnectorSmokeTest.java`.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeRegisterTableProcedureTest.java [hunk 6]
```diff
@@ -177,6 +177,29 @@ public void testRegisterTableWithDifferentTableName()
         assertUpdate(format("DROP TABLE %s", tableNameNew));
     }
 
+    @Test
+    public void testRegisterTableWithTrailingSpaceInLocation()
+    {
+        String tableName = "test_create_table_with_trailing_space_" + randomNameSuffix();
+        String tableLocationWithSpace = tableName + " ";
+
+        assertQuerySucceeds(format("CREATE TABLE %s WITH (location = '%s') AS SELECT 1 as a, 'INDIA' as b, true as c", tableName, tableLocationWithSpace));
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeRegisterTableProcedureTest.java:186]: nit: Uppercase `as`.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java [hunk 7]
```diff
@@ -565,6 +565,19 @@ public void testPathColumn()
         }
     }
 
+    @Test
+    public void testTableLocationTrailingSpace()
+    {
+        String tableName = "table_with_space";
```
Comment (krvikash) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java:571]: append with `randomNameSuffix`

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java [hunk 8]
```diff
@@ -565,6 +565,19 @@ public void testPathColumn()
...
+        String tableLocationWithSpace = tableName + " ";
```
Comment (krvikash) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java:572]: I think you can inline `tableLocationWithSpace` OR construct fully qualified path here like `String tableLocationWithTrailingSpace = "s3://" + bucketName + "/" + tableName + " ";`

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeRegisterTableProcedureTest.java [hunk 9]
```diff
@@ -177,6 +177,27 @@ public void testRegisterTableWithDifferentTableName()
         assertUpdate(format("DROP TABLE %s", tableNameNew));
     }
 
+    @Test
+    public void testRegisterTableWithTrailingSpaceInLocation()
+    {
+        String tableName = "test_create_table_with_trailing_space_" + randomNameSuffix();
```
Comment (alexjo2144) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeRegisterTableProcedureTest.java:183]:
```suggestion
        String tableName = "test_register_table_with_trailing_space_" + randomNameSuffix();
```

**Review by alexjo2144 — APPROVED** [review]

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeRegisterTableProcedureTest.java [hunk 10]
```diff
@@ -177,6 +177,27 @@ public void testRegisterTableWithDifferentTableName()
...
+        String tableLocationWithTrailingSpace = dataDirectory + "/"+ tableName + " ";
```
Comment (krvikash) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeRegisterTableProcedureTest.java:184]: missing space `+ "/" +`

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java [hunk 11]
```diff
@@ -1827,6 +1827,29 @@ public void testRegisterTable()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testRegisterTableWithTrailingSpaceInLocation()
+    {
+        String tableName = "test_create_table_with_trailing_space_" + randomNameSuffix();
+        String tableLocationWithSpace = bucketUrl() + tableName + " ";
```
[the remark this answers is not in the document]
Comment (krvikash) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java:1834]: Gentle reminder.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java [hunk 12]
```diff
@@ -565,6 +565,19 @@ public void testPathColumn()
         }
     }
 
+    @Test
+    public void testTableLocationTrailingSpace()
+    {
+        String tableName = "table_with_space_" + randomNameSuffix();
+        String tableLocationWithSpace = "s3://" + bucketName + "/" + tableName + " ";
```
Comment (krvikash) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java:572]: nit: `tableLocationWithSpace` -> `tableLocationWithTrailingSpace`

**Review by krvikash — APPROVED** [review]
LGTM. Few comments.

**Comment by ebyhr** [comment]
/test-with-secrets sha=7440fda8233d45accf42a91d10d5d188256c61ed

**Comment by ebyhr** [comment]
/test-with-secrets sha=7a75980e67837ed68c51019cd547c259ae257b56

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
CI with secrets is running in https://github.com/trinodb/trino/actions/runs/4320615128/jobs/7541019227

**Pull request merged**